### PR TITLE
Refactor License Scout

### PR DESCRIPTION
**Pull request by tduffield**
This is a major refactor of License Scout that modifies it's operating
model to drive more configuration from external configuration managed
by YAML files rather than by configuration stored within the gem itself.

This allows teams to add exceptions, fallbacks, and other configuration
based on their needs without requiring us to release a new version of
the license_scout gem.

Signed-off-by: Tom Duffield <tom@chef.io>

### CODE REVIEW

File: lib/license_scout/dependency_manager/bundler.rb [hunk 1]
```diff
@@ -42,13 +52,31 @@ def detected?
         File.exist?(gemfile_path) && File.exist?(lockfile_path)
       end
 
+      def dependencies
+        dependency_data.map do |gem_data|
+          dep_name = gem_data["name"]
+          dep_version = gem_data["version"]
+
+          dep_path = if dep_name == "bundler"
```
Comment (danielsdeleo) [inline lib/license_scout/dependency_manager/bundler.rb]: Probably worth keeping the comments explaining the special cases

File: lib/license_scout/dependency_manager/habitat.rb [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+      def license_from_manifest(manifest_content)
+        /^*\s+__License__:\s+(.+)$/.match(manifest_content)[1].strip.split("\s")
+      end
+
+      def pkg_deps
+        @pkg_deps ||= begin
+          plan_path = File.exist?(plan_sh_path) ? plan_sh_path : habitat_plan_sh_path
+
+          c = Mixlib::ShellOut.new("source #{plan_path}; echo $pkg_deps", LicenseScout::Config.environment)
```
Comment (danielsdeleo) [inline lib/license_scout/dependency_manager/habitat.rb]: What version of bash did you try this with? I had to do something like `echo ${pkg_deps[*]}`

Comment (tduffield) [reply lib/license_scout/dependency_manager/habitat.rb]: That's a good question - I just ran it locally. It's probably worth me circling back to that.

File: spec/fixtures/habitat/plan.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+pkg_name=fixture
+pkg_origin=chef
+pkg_version="0.1.0"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Apache-2.0")
+pkg_deps=(
+  core/glibc
```
Comment (danielsdeleo) [inline spec/fixtures/habitat/plan.sh]: You should put 2 or 3 licenses in here 😉

Comment (tduffield) [reply spec/fixtures/habitat/plan.sh]: I suppose I could. I know it works since I've run it on other more robust projects, but I will update the tests.

**Pull request merged**